### PR TITLE
Dashboard Edited/Saved notifications leak into other views

### LINKED ISSUE #37306: Dashboard Edited/Saved notifications leak into other views
**What happened**:
When I:

In Tab A:
 - Open up a dashboard
 - Navigated to another page (in my case, `/alerting/routes` with ng alerting enabled)

In Tab B
 - Edit the dashboard I was viewing in Tab A and save

I see "Dashboard Saved" (or in another case "someone modified the dashboard" modal) in Tab A even though I am no longer on the dashboard that was edited.

**What you expected to happen**:
Only to get these notifications while actively in a dashboard.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Grafana version: ddf5b65c51
- Data source type & version:
- OS Grafana is installed on:
- User OS & Browser:
- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by gitstart**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #37306

**Demo**
https://www.loom.com/share/9f30bcb9ac2c4851ade0834552413445

### CODE REVIEW

**Comment by gitstart** [comment]
Hi @kylebrandt, kindly review the PR 🙏

File: public/app/features/dashboard/containers/DashboardPage.tsx [hunk 1]
```diff
@@ -118,6 +118,7 @@ export class UnthemedDashboardPage extends PureComponent<Props, State> {
   }
 
   componentWillUnmount() {
+    dashboardWatcher.leave();
```
Comment (torkelo) [inline public/app/features/dashboard/containers/DashboardPage.tsx:121]: this is better done from cleanUpDashboardAndVariables

**Comment by gitstart** [comment]
Hi @torkelo! Thank you for the review. Requested changes have been made, kindly look at it

**Comment by gitstart** [comment]
Hey @torkelo! Kindly review the PR 🙏

**Review by torkelo — APPROVED** [review]

**Comment by gitstart** [comment]
Hi @torkelo, please do we need more reviews for this PR to be merged?

**Pull request merged**